### DISCUSSION TITLE
MoLang Compilation

### DISCUSSION THREAD

**Discussion by kennyvv** [ConcreteMC/Alex#118] [Ideas]
## The problem
Currently calculating model animations takes up a lot of time, this is mainly caused by the execution of our MoLang scripts. 

### Ideas
We can "compile" our molang statements to .net functions. This way, we don't need to evaluate every molang statement by itself anymore and we can instead just execute the generated code. Stuff like query access doesn't need to lookup a dictionary anymore and can instead directly access the property on the entity. This would save a ton of execution time. 

### Implementation
I'm not yet sure on how exactly we should implement this. More research is required.

**Comment by TruDan** [comment]
Dynamic/Expando object / anonymous class instead of dictionary?
